### PR TITLE
Add tasks as a top level navigation

### PR DESCRIPTION
**Pull request by steveodonovan**
@AlanGreene A little wide reaching and large here. Can break this down if you like. The overall design is also still under review in https://github.com/tektoncd/dashboard/issues/12

There are a few pieces here which will be a little cleaner when we have the store in place.

### CODE REVIEW

File: src/containers/Home/Home.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        <StructuredListWrapper border selection>
+          <StructuredListHead>
+            <StructuredListRow head>
+              <StructuredListCell head>Task</StructuredListCell>
+            </StructuredListRow>
+          </StructuredListHead>
+          <StructuredListBody>
+            <StructuredListRow className="pipeline" key="pipeline">
+              <StructuredListCell>
+                <Link to="/pipelines">pipelines</Link>
```
Comment (AlanGreene) [inline src/containers/Home/Home.js]: link text: Pipelines, similarly Tasks below

File: src/containers/Home/Home.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+      <main>
+        <StructuredListWrapper border selection>
+          <StructuredListHead>
+            <StructuredListRow head>
+              <StructuredListCell head>Task</StructuredListCell>
+            </StructuredListRow>
+          </StructuredListHead>
+          <StructuredListBody>
+            <StructuredListRow className="pipeline" key="pipeline">
```
Comment (AlanGreene) [inline src/containers/Home/Home.js]: `key` not necessary here

File: src/containers/PipelineRun/PipelineRun.js [hunk 3]
```diff
@@ -17,12 +17,17 @@ import { InlineNotification } from 'carbon-components-react';
...
 
-import PipelineRunHeader from '../../components/PipelineRunHeader';
+import RunHeader from '../../components/RunHeader';
 import StepDetails from '../../components/StepDetails';
 import TaskTree from '../../components/TaskTree';
-import { getStatus } from '../../utils';
+import {
+  getStatus,
+  selectedTask,
+  taskRunStep,
```
Comment (AlanGreene) [inline src/containers/PipelineRun/PipelineRun.js]: nit: order imports alphabetically for easier scanning

File: src/containers/TaskRuns/TaskRuns.js [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+    let notification;
+    const tasks = await getTasks();
+    const task = tasks.find(
+      currentTask => currentTask.metadata.name === selectedTaskName
+    );
+    let taskRuns = await getTaskRuns();
+    taskRuns = taskRuns
+      .filter(
+        taskRun =>
+          taskRun.spec.taskRef && taskRun.spec.taskRef.name === selectedTaskName
```
Comment (AlanGreene) [inline src/containers/TaskRuns/TaskRuns.js]: TODO: discuss additional back end API to retrieve TaskRuns for given Task name.
Not required for this initial PR but should follow up separately.

File: src/containers/TaskRuns/TaskRuns.js [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+        id: step.name
+      };
+    });
+    return steps;
+  }
+
+  // once redux store is available errors will be handled properly with dedicated components
+  notification(notification) {
+    const { kind, message } = notification;
+    const titles = {};
```
Comment (AlanGreene) [inline src/containers/TaskRuns/TaskRuns.js]:
```
const titles = {
  info: '...',
  error: '...'
};
```

TODO: 
Also might re-word the info text. 'Task runs not available' could sound like an error.
There's a similar message above: https://github.com/tektoncd/dashboard/pull/49/files#diff-ccd95d56ab624312e691721ff80a1260R101 'Task has never run'.

Let's revisit this as part of the design update.

File: src/utils/index.js [hunk 6]
```diff
@@ -35,3 +35,31 @@ export function getPipelineRunStatusIcon({ reason, status }) {
...
+    return {};
+  }
+
+  const { id, stepName, stepStatus, status, reason, ...definition } = step;
+
+  return {
+    definition,
+    reason,
+    stepName,
+    stepStatus,
```
Comment (AlanGreene) [inline src/utils/index.js]: What's the difference between `status` and `stepStatus` here?

Comment (steveodonovan) [reply src/utils/index.js]: Step status is actually the value for the step in the task run (
```
    {
      terminated: {
        exitCode: 1,
        'Error',
        startedAt: '2019',
        finishedAt: '2019',
        containerID: 'containerd://testid'
      }
    }
```
for example

status is the actual succeeded/failed/waiting ...

Bad naming for sure but preexisting. Changing it in this PR might mean hitting other components. Mind if I address it separately ?

Comment (AlanGreene) [reply src/utils/index.js]: ah yes, ok let's follow up before this gets forgotten 👍

**Comment by steveodonovan** [comment]
@AlanGreene Could I get another once over please ? Thanks

**Comment by AlanGreene** [comment]
The actual change isn't as big as the diff suggests, ~50% tests and refactoring by moving / extraction.

Verified the changes locally, including at smaller screen sizes. It's consistent with the existing design and layout 👍 Core components are now more easily reusable too.

/approve

**Pull request merged**